blog/_posts/2023-04-06-march-2023-scala-tooling-summit.md

```diff
@@ -0,0 +1,357 @@
+gone ahead and formalized the [team
+structure](https://github.com/build-server-protocol/build-server-protocol/pull/465)
+a bit, are in the process of migrating to a
+[Smithy](https://smithy.io/2.0/index.html) as a [source of
+truth](https://github.com/build-server-protocol/build-server-protocol/pull/426),
+and are actively exploring a smoother [BSP
+Discovery](https://build-server-protocol.github.io/docs/overview/server-discovery)
+process.
+
+### Making tooling more robust
+
+Having your tooling not work as expected is frustrating. Not knowing why it’s
+not working is even more frustrating. This conversation surrounds ideas and
+topics like:
+
+- Better logging
+- Telemetry
+- Fault tolerance
+- Stress testing
+
+As a result of these conversations the Metals team will be exploring usage of
+[Sentry](https://sentry.io/welcome/) to better track when things are going
+wrong. The idea here would be to provide an experience similar to IntelliJ’s
+[Exception Analyzer](https://blog.jetbrains.com/yole/archives/000072.html).
+[LSP](https://microsoft.github.io/language-server-protocol/) also has a
+[telemetry](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#telemetry_event)
+event that could be utilized, but more research needs to be done here.
+
+### Limiting the ways to import a project into your IDE
+
+Both Metals and IntelliJ users have multiple ways to import projects. This could
+be simply choosing your build server in Metals, or choosing to use BSP at all in
+IntelliJ. There are ongoing discussions and work going on in both of these tools
+to make the getting started experience on a project more seamless no matter the
+tool or project. Here are a couple links related to this:
+
+- [Metals default build server
+  discussion](https://github.com/scalameta/metals/discussions/4505#discussioncomment-4571546)
+- [New BSP focused plugin from
+  IntelliJ](https://lp.jetbrains.com/new-bazel-plugin/) which may be replacing
+  the current BSP implementation.
+
+### Porting Scalameta to Scala 3
+
+[Scalameta](https://scalameta.org/) is a core part of the tooling ecosystem
+being utilized by tools like Metals, Scalafmt, Scalafix, and mdoc. While
+Scalameta can handle parsing Scala 3 code, it can’t be used as a library in
+Scala 3 projects. The main hiccup in this effort is the extensive use of macros,
+which aren’t easily migrated. The main outcome of this discussion was simply
+better understanding the challenges involved, so that if someone is willing to
+put in the time here, we have a starting point and some research done.
+
+### IDE support in “polyglot” projects that contain Scala
+
+Traditionally IntelliJ has been the leader in this front offering great support
+for mixed JVM-language projects. However this is made trickier with tools that
+may not always be well-supported in this context like sbt or Mill. With the
+upcoming new [BSP plugin](https://lp.jetbrains.com/new-bazel-plugin/) for
```

Comment:
I'd prefer not linking it publicly yet

Reply:
Alright, just to make sure can I still mention there is a new BSP plugin coming?

Reply:
yes, that's fine